### PR TITLE
Added server side paging

### PR DESCRIPTION
**Pull request by bryanwong8**
Loom with DevExtreme DataGrid package:
https://www.loom.com/share/9c61bbac5e8a4f43a5cd4353d05df597?sharedAppSource=team_library

Loom without DevExtreme DataGrid package:
https://www.loom.com/share/f21f4b08c98d4bd48a4737ddbc40ab08?sharedAppSource=team_library

Loom with using a pagination class:
https://www.loom.com/share/bfdf06a816f04634aa56d516c16cc2c3?sharedAppSource=team_library

- Implemented server side pagination with DevExtreme
- Cleaned up code by replacing hard coded values with enums
- Replaced hand made date parser with date-fns

EDIT:
As @uskudnik mentioned, I needed to do the project in a more DRF way. So I replaced the DevExtreme model with custom filters, ordering, and pagination.

### CODE REVIEW

**Review by zenput-paul — CHANGES_REQUESTED** [review]

**Review by uskudnik — COMMENTED** [review]
LGTM, the biggest thing to look at is what @zenput-paul mentioned about pagination.

File: frontend/src/components/Modal.tsx [hunk 1]
```diff
@@ -84,14 +87,14 @@ const CustomModal = (props: ModalProps) => {
                 {activeItem.priority}
               </DropdownToggle>
               <DropdownMenu>
-                <DropdownItem onClick={() => handleDropdownChange("Low")}>Low</DropdownItem>
-                <DropdownItem onClick={() => handleDropdownChange("Mid")}>Mid</DropdownItem>
-                <DropdownItem onClick={() => handleDropdownChange("High")}>High</DropdownItem>
+                <DropdownItem onClick={() => handleDropdownChange(priority.LOW)}>Low</DropdownItem>
```
[the remark this answers is not in the document]
Comment (bryanwong8) [reply frontend/src/components/Modal.tsx]: Addressed in latest push

File: backend/backend/settings.py [hunk 2]
```diff
@@ -130,4 +130,8 @@
 
 CORS_ORIGIN_WHITELIST = (
     'http://localhost:3000',
-)
\ No newline at end of file
+)
+
+REST_FRAMEWORK = {
+    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.LimitOffsetPagination'
+}
```
Comment (zenput-paul) [inline backend/backend/settings.py:137]: Don't think you need this, this is to set up the LimitOffsetPagination as the pagination class for all your viewsets. Don't think it hurts to leave it, but just so you know that you don't need it to have the pagination in the ToDoViewset

**Review by zenput-paul — APPROVED** [review]
LGTM! 💯

File: backend/backend/settings.py (hunk 2 above)
Comment (bryanwong8) [reply backend/backend/settings.py:137]: Good catch

**Pull request merged**